### ISSUE TITLE
Excruciating lag when using Vysor with Zoom

### ISSUE THREAD

**Issue by elisajdu** [koush/vysor.io#935]
Not sure if this is a Vysor issue or Zoom issue.

1. With every program running (Zoom, PowerPoint, Vysor), the tablet lag is a few seconds. Perfectly acceptable.
2. With Zoom capturing Vysor, the lag increases by a factor of ... ten? I have it recorded.
3. If I use the screen capture function in Zoom, 
 - if Zoom is capturing Vysor, the lag is huge. 
 - if Zoom is capturing anything else, the lag is 'normal'.

#3 is my work-around. I write in Vysor, go to the PowerPoint for a few seconds, then go back and what I have written is now showing. This is inconvenient, but it works.

(Bug report: https://gist.github.com/VysorLogs/413683a7f4396072c38d62636ba869e2)

**Comment by koush** [comment]
Will take a look

**Comment by parvezsyed** [comment]
> Not sure if this is a Vysor issue or Zoom issue.
> 
> 1. With every program running (Zoom, PowerPoint, Vysor), the tablet lag is a few seconds. Perfectly acceptable.
> 2. With Zoom capturing Vysor, the lag increases by a factor of ... ten? I have it recorded.
> 3. If I use the screen capture function in Zoom,
> 
> * if Zoom is capturing Vysor, the lag is huge.
> * if Zoom is capturing anything else, the lag is 'normal'.
> 
> #3 is my work-around. I write in Vysor, go to the PowerPoint for a few seconds, then go back and what I have written is now showing. This is inconvenient, but it works.
> 
> (Bug report: https://gist.github.com/VysorLogs/413683a7f4396072c38d62636ba869e2)

Saw the same lag issue with google meet on mac. Any conference app does this I presume.

**Comment by koush** [comment]
Weird, I am not detecting or blocking any virtual cam stuff. This doesn't happen on Mac for me. I'll check Windows. Did you try the Vysor Virtual camera? That's exactly what this is built for: sharing your vysor window.

**Comment by parvezsyed** [comment]
how to use the vysor virtual camera? I use the vysor app on mac and click on the play button when my ipad is detected on my mac. connection happens through bluetooth?

**Comment by elisajdu** [comment]
> Weird, I am not specially handling or blocking any capturing stuff. Did you try the Vysor Virtual camera? That's exactly what this is built for: sharing your vysor window.

More information: the lag correlates with the number of active processes. Shutting down everything that accesses the CPU decreases the lag. But even then, with only Zoom, Vysor, and Powerpoint running, the lag is about 20 %. So after a minute of writing, Vysor is around 12 seconds behind. My workaround is to wiggle the USB connection to cause a reset, which updates to the final image after reconnection.

My system: HP zBook G2 • i7-4710MQ 2.5 GHz • Windows 7 Pro • 24 GB RAM • 450 GB free HD.

**Comment by parabellum2000** [comment]
I've also experienced this issue. 

Pixel 3 and Pixel 6 Pro
Macbook Pro 2020 16gb RAM i7

I used the free version for several months before upgrading to the paid version. Only with the paid version, did I have the problem. I tried multiple different codecs. It didn't help.  What worked for me was turning the high-quality streaming back down to low quality. On low-quality screen sharing looks just fine. The timing stays in sync. I think it's a process issue. I'm not sure on which side Mac or Android. @koush Thanks for maintaining this software!

**Comment by koush** [comment]
can you try this? 

https://osherove.com/blog/2020/8/18/solved-high-cpu-and-lagging-when-screen-sharing-in-zoom-2020-multiple-screens

**Comment by koush** [comment]
I do not think this is a Vysor issue. Vysor has no awareness of Zoom. It is likely an issue with Zoom's screen capture being poorly implemented.

Vysor works fine on it's own, and only when Zoom captures it does it get laggy: that's on Zoom, and very unlikely anything I could fix.

**Comment by parabellum2000** [comment]
I actually experience the issue using Google Meet. I don’t think it has similar settings to Zoom in this regard.

**Comment by koush** [comment]
> I actually experience the issue using Google Meet. I don’t think it has similar settings to Zoom in this regard.
> […](#)

Have you tried using the Vysor camera feature? it creates a virtual camera of your android/iOS screen.

**Comment by elisajdu** [comment]
@koush 

Changing the bitrate does not significantly affect the lag. 
Having more programs dramatically affects the lag.
**_Changing the decoder DRAMATICALLY affects the lag._** 

20 seconds of continuous scribble took the following times to render.
 
Recorded at 2 Mbps
     **Native hardware: 21 seconds**
     Webcodec software: 30 seconds

Recorded at 8 Mbps
     Native software: 118 seconds
     **Native hardware: 21 seconds**
     Webcodec software: 35 seconds
     Webcodec hardware: 57 seconds

Recorded at 16 Mbps
     **Native hardware: 21 seconds**
     Webcodec software: 42 seconds
